pydicom/filebase.py

```diff
@@ -30,19 +30,13 @@ def __init__(self, *args, **kwargs):
     def read_le_tag(self):
         """Read and return two unsigned shorts (little endian)
            from the file."""
-        bytes_read = self.read(4)
-        if len(bytes_read) < 4:
-            # needed for reading "next" tag when at end of file
-            raise EOFError
+        bytes_read = self.read(4, need_exact_length=True)
         return unpack(b"<HH", bytes_read)
 
     def read_be_tag(self):
```

Comment:
Or, if this is not used anyway in the user code (it is indeed very unlikely) we can just add a note in the porting guide instead of adding a deprecation.